### PR TITLE
[labs/cli] Lazily install and locally version localize

### PR DESCRIPTION
**Pull request by rictic**
Fixes https://github.com/lit/lit/issues/2868

### CODE REVIEW

File: packages/labs/cli/src/test/gen/react_test.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ test('basic wrapper generation', async ({rig, testConsole}) => {
   testConsole.alsoLogToGlobalConsole = true;
   await cli.run();
 
-  assert.equal(testConsole.errorStream.buffer.join(''), '');
+  assert.snapshot(testConsole.errorStream.text, '');
```
Comment (rictic) [inline packages/labs/cli/src/test/gen/react_test.ts:58]: This generates a better error message

File: packages/labs/cli/src/lib/commands/localize.ts [hunk 2]
```diff
@@ -4,53 +4,12 @@
...
-    console.error(
-      'Use one of the localize subcommands, like `lit localize build` or ' +
-        '`lit localize extract`. Run `lit help localize` for more help.'
-    );
-    return {
-      exitCode: 1,
-    };
-  },
+  importSpecifier: '@lit/localize-tools/lib/cli-command.js',
+  installFrom: '@lit/localize-tools',
```
[the remark this answers is not in the document]
Comment (rictic) [reply packages/labs/cli/src/lib/commands/localize.ts:14]: How will changeset deal with this? Would we have to override it if a change is breaking for the library but fine for the CLI?

File: packages/labs/cli/src/test/uvu-wrapper.ts [hunk 3]
```diff
@@ -47,9 +47,12 @@ function timeout<T>(
         );
       }),
     ]);
-    result.finally(() => {
-      clearTimeout(timeoutId);
-    });
+    result
```
[the remark this answers is not in the document]
Comment (rictic) [reply packages/labs/cli/src/test/uvu-wrapper.ts]: +1, done

File: packages/labs/cli/src/lib/commands/localize.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rictic) [reply packages/labs/cli/src/lib/commands/localize.ts]: Done

File: packages/labs/cli-localize/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * Defines our interface within the Lit CLI.
+ */
+export const getCommand = () => {
```
Comment (justinfagnani) [inline packages/labs/cli-localize/src/index.ts:10]: It'd be nice to get type-checking here. Are circular deps that tricky for npm?

File: packages/labs/cli-localize/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
```
Comment (justinfagnani) [inline packages/labs/cli-localize/tsconfig.json:2]: We should have a common Node tsconfig we extend from...

**Review by justinfagnani — APPROVED** [review]

File: packages/labs/cli-localize/package.json [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  "wireit": {
+    "build": {
+      "dependencies": [
+        "build:deps",
+        "build:compile"
+      ]
+    },
+    "build:compile": {
+      "command": "tsc --skipLibCheck || echo ''",
+      "#comment": "This never fails and always emits output so that we can run tests of code that doesn't type check",
```
[the remark this answers is not in the document]
Comment (rictic) [reply packages/labs/cli-localize/package.json:28]: It serves two purposes:

1. Lets us compile in parallel, with no waterfall. If A depends on B depends on C then we have to compile C then we can compile B then we can compile A. With this layout we can compile all of them at the same time.
2. We can run tests while there are type errors. We have a lot of very strict compiler flags set up, but when developing it's useful to be able to run a test even when e.g. there's an unused local variable.

**Review by rictic — COMMENTED** [review]
Doing the types right and introducing a cycle unfortunately broke the windows tests. Merging

**Pull request merged**